app/models/manageiq/providers/amazon/inventory/parser/cloud_manager.rb

```diff
@@ -4,10 +4,15 @@ class ManageIQ::Providers::Amazon::Inventory::Parser::CloudManager < ManageIQ::P
   def parse
     log_header = "MIQ(#{self.class.name}.#{__method__}) Collecting data for EMS name: [#{collector.manager.name}] id: [#{collector.manager.id}]"
     $aws_log.info("#{log_header}...")
+
     # The order of the below methods does matter, because they are searched using find instead of lazy_find
     flavors
 
+    # @mapper has to be initialized for parsing images & instances
+    @tag_mapper = ContainerLabelTagMapping.mapper
+
     # The order of the below methods doesn't matter since they refer to each other using only lazy links
+    persister.collections[:tags_to_resolve] = @tag_mapper.tags_to_resolve_collection
```

Comment:
@cben ok, so continuing from https://github.com/ManageIQ/manageiq-providers-amazon/pull/382#discussion_r159463356

lets move this to persister init, we can put shared method to ManageIQ::Providers::Amazon::Inventory::Persister and just call it in Cloud and TargetCollection Persisters for now

And we should there add a TODO for perf, that this will always load all ContainerLabelTagMapping (or other?). But this table is fairly small? It's the Tagging that is huge, right?

Reply:
Mapper init is reading ContainerLabelTagMapping.all.  It's fairly small, 1 per mapping defined in UI, independent of inventory size and number of tags created/assigned.
Moving...

Reply:
was about to ask about TargetCollection then saw you already anticipated that :-)
Done, PTAL.